### PR TITLE
add api-docs build step

### PR DESCRIPTION
**Pull request by b-admike**
Add the `build:apidocs` build step to our `@loopback/cli` and `@loopback/build` packages.
Blocks https://github.com/strongloop-internal/apidocs.strongloop.com/pull/81.

## Checklist

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] Related API Documentation was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `packages/example-*` were updated

### CODE REVIEW

File: packages/build/package.json [hunk 1]
```diff
@@ -33,15 +33,20 @@
...
+    "prepare": "npm run build:apidocs",
     "prettier:cli": "node bin/run-prettier \"bin/**/*.js\"",
     "prettier:check": "npm run prettier:cli -- -l",
     "prettier:fix": "npm run prettier:cli -- --write",
     "test": "npm run mocha",
     "mocha": "node bin/select-dist mocha --timeout 30000 --opts test/mocha.opts \"test/integration/*.js\"",
     "posttest": "npm run lint"
   },
+  "files": [
+    "api-docs"
```
Comment (bajtos) [inline packages/build/package.json]: IIUC, this means that `npm publish` will include only api-docs files, but not the actual build helpers! See https://docs.npmjs.com/files/package.json#files

I think you should either revert this addition of a "file" field, or else make sure all important files/directories are listed in "files".

**Comment by raymondfeng** [comment]
As a side note, `@loopback/build` and `@loopback/cli` does not expose public APIs. Do we still want to run apidocs generation?

File: packages/build/package.json [hunk 2]
```diff
@@ -33,15 +33,20 @@
     "lb-clean": "./bin/run-clean.js"
   },
   "scripts": {
+    "build:apidocs": "lb-apidocs",
```
Comment (raymondfeng) [inline packages/build/package.json]: It should be `"build:apidocs": "node bin/generate-apidocs"`.

**Comment by b-admike** [comment]
>As a side note, @loopback/build and @loopback/cli does not expose public APIs. Do we still want to run apidocs generation?
Yeah I don't think it's worth having apidocs generation for build and cli. @shimks also tried it out, and it seems only their READMEs were exposed.

**Pull request closed without merging**